be/src/storage/rowset/segment_v2/storage_page_decoder.h

```diff
@@ -0,0 +1,70 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#pragma once
+
+#include <memory>
+
+#include "common/status.h"
+#include "gen_cpp/segment_v2.pb.h"
+#include "storage/rowset/segment_v2/common.h"
+#include "util/slice.h"
+
+
+namespace starrocks {
+
+namespace segment_v2 {
+
+class DataDecoder {
+public:
+    DataDecoder() = default;
+    virtual ~DataDecoder() = default;
+
+    static DataDecoder* get_data_decoder(EncodingTypePB encoding);
+
+    virtual void reserve_head(uint8_t head_size) {}
+
+    virtual Status decode_page_data(PageFooterPB* footer, uint32_t footer_size, EncodingTypePB encoding,
+                                    std::unique_ptr<char[]>* page, Slice* page_slice) {
+        return Status::OK();
+    }
+};
+
+class BitShuffleDataDecoder : public DataDecoder {
```

Comment:
Place the class declaration in .cpp file too

Reply:
> Place the class declaration in .cpp file too

done, PTAL